### ISSUE TITLE
can't fetch link database from two devices

### ISSUE THREAD

**Issue by rspier** [abates/insteon#14]
There's a high probability that I'm missing something obvious.... but I can't seem to fetch the link database from two devices while using the same *plm.PLM twice.

I think this used to work (many many months ago while we were debugging the timeout issues in #2), and it's not obvious to me what changed.  Or maybe my PLM is just wonky.

Simple reproduction: [main.go](https://github.com/abates/insteon/files/3895040/main.go.txt)

Output:
``` shell
$ go run . --logtostderr --port /dev/ttyUSB1
modem opened as PLM (34.78.d9)
I1126 19:58:21.011128   14421 main.go:41] fetch(33.ad.00)
2019/11/26 19:58:21 DEBUG Retrieving Device link database
found 6 links on 33.ad.00
I1126 19:58:34.687887   14421 main.go:41] fetch(33.14.1d)
E1126 19:58:37.688169   14421 main.go:74] error connecting to 33.14.1d: Ack Timeout
I1126 19:58:47.688381   14421 main.go:41] fetch(2c.7d.24)
E1126 19:58:50.688690   14421 main.go:74] error connecting to 2c.7d.24: Ack Timeout
```

The order of the addresses doesn't matter.  (i.e. it's not a problem with a specific device, or specific type of device.)  Adding time.Sleep() calls doesn't help.

**Comment by abates** [comment]
Confirmed behavior on my end.  I will try to troubleshoot soon, but I'm pretty busy with family activities this week.

**Comment by rspier** [comment]
No rush.  Thanks for looking into it whenever you get to it.  Happy Thanksgiving! :turkey: 

-R

**Comment by abates** [comment]
Alright, so finding the problem was pretty simple, and it's a complete facepalm.  Fixing it will take a little thought.  The PLM code has a read loop that runs in a go routine.  This reads packets coming in from the serial port and dispatches them to either the PLM message channel (for PLM related operations like adding or removing PLM links or changing PLM config) or (in the case of an Insteon message) to the insteon receive channel.  There is only one insteon receive channel, and each time you call Open() each devices' connection reads from the same channel.

This introduces a race condition.  In the case of the sample main, each of the three devices has a running read loop.  All three devices read from the PLM's single insteon message channel.  Therefore, once one reader consumes the message from the channel, the others will time out waiting.

The fix for this could be one of two things:

1) Add a new go routine in the PLM code that fans out insteon messages to all connections.  The filtering logic is implemented in insteon.Connection so it would be a simple for loop to fan out

2) Add a synchronized map of receive channels to the PLM, one per open connection and have the receive loop only deliver to the connection for the matching source address. 

When I set this project aside earlier this year I had started to look at revamping some of the PLM code.  It is the least tested part of this project and could use some TLC.  I'll think this over and get back to it in a day or so.
